src/Hazelcast.Net.Linq.Async/HKeyValuePair.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Hazelcast.Linq
+{
+    /// <summary>
+    /// A key value pair for <see cref="IHMap"/> entries on LINQ.
+    /// </summary>
+    /// <typeparam name="TKey">Type of Key</typeparam>
+    /// <typeparam name="TValue">Type of Value</typeparam>
+    public struct HKeyValuePair<TKey, TValue>
```

Comment:
Why do we need this alternative to `KeyValuePair`? Also, it's a non-readonly `struct`. Feels dangerous but there has to be a reason which we should at least document in comments.